pkg/config/validation/validation.go

```diff
@@ -3458,3 +3460,57 @@ func validateTelemetryProviders(providers []*telemetry.ProviderRef) error {
 	}
 	return nil
 }
+
+// ValidateWasmPlugin validates a WasmPlugin.
+var ValidateWasmPlugin = registerValidateFunc("ValidateWasmPlugin",
+	func(cfg config.Config) (Warning, error) {
+		spec, ok := cfg.Spec.(*extensions.WasmPlugin)
+		if !ok {
+			return nil, fmt.Errorf("cannot cast to service entry")
+		}
+
+		errs := Validation{}
+		errs = appendValidation(errs,
+			validateWorkloadSelector(spec.Selector),
+			validateWasmPluginURL(spec.Url),
+			validateWasmPluginSHA(spec),
+		)
+		return errs.Unwrap()
+	})
+
+func validateWasmPluginURL(pluginURL string) error {
+	if pluginURL == "" {
+		return fmt.Errorf("url field needs to be set")
+	}
+	validSchemes := map[string]bool{
+		"": true, "file": true, "http": true, "https": true, "oci": true,
```

Comment:
I imagine if "file" is given, we would hit https://github.com/istio/istio/blob/master/pkg/wasm/cache.go#L159?

Reply:
no- because if the user sets a `file://` URL, we'll generate a `local` datasource for the HTTP filter, which the wasm conversion code will ignore (it only converts `remote` datasources)

Reply:
see https://github.com/istio/istio/blob/74d1c06903ded69df9f090b1e3fe88fd8686804d/pilot/pkg/model/extensions.go#L73-L83

Reply:
oh I see! Thanks!